Libraries/StyleSheet/ColorPropType.js

```diff
@@ -9,26 +9,31 @@
   * @providesModule ColorPropType
   */
 'use strict';
+
 var ReactPropTypes = require('ReactPropTypes');
+
 var tinycolor = require('tinycolor');
 
-var colorValidator = function (props, propName) {
+var ColorPropType = function(props, propName) {
   var selectedColor = props[propName];
-  if (selectedColor === null || selectedColor === undefined || selectedColor.toString().trim() === '') {
-    return new Error(
-      `Invalid argument supplied to ${propName}.Expected a string like #123ADF or 'red'.`
-    );
+  if (selectedColor === undefined) {
+    return;
   }
 
-  if (tinycolor(selectedColor.toString().trim()).isValid()) {
-    return null;
+  if (tinycolor(selectedColor) === null) {
+    return new Error(
+`Invalid color supplied to ${propName}: ${selectedColor}. Valid color formats are
+  - '#f0f' (#rgb)
+  - '#ff00ff' (#rrggbb)
+  - '#ff00ff00' (#aarrggbb)
```

Comment:
We only expose it because of a flaw in our propType validation which means that we use the same validator for the public prop as for the native prop, which means we have no way to expose `View.backgroundColor` as a string if the underlying `RCTView.backgroundColor` is an int, unless we rename the native property. This is somewhat clowny, but I've not had time to find a better workaround.

Allowing end-users to specify colors as hex ints is kinda nice, but it's also a bit fragile because we can't support both rgb and rgba that way, e.g.

    0xff0000

could be interpreted as either red or green depending on whether we assume it's an rgb or rgba literal. argb format doesn't have this problem, so I suggest we either keep the int literal support but keep it as argb (which is different from the string format, but does that really matter?), or drop support for it and fix the propType validator.